plugins/modules/vmware_guest.py

```diff
@@ -1284,13 +1289,15 @@ def sanitize_cdrom_params(self):
                     if cdrom_spec['type'] == 'iso' and not cdrom_spec.get('iso_path'):
                         self.module.fail_json(msg="cdrom.iso_path is mandatory when cdrom.type is set to iso.")
 
```

Comment:
I suggest we add a deprecation warning (e.g: https://github.com/ansible-collections/community.general/blob/main/plugins/cache/redis.py#L81-L83) when the `is_list` is false and `controller_number` and `unit_number` are not set.
We are at 1.0.0 currently, and we will release a version every month. So I guess we can target 1.18 (18 months).

Reply:
FYI, @Akasurde pushed a PR for that: https://github.com/ansible-collections/vmware/pull/317